src/state/apis/nft/selectors.ts

```diff
@@ -1,13 +1,40 @@
+import { QueryStatus } from '@reduxjs/toolkit/query'
 import type { AccountId, AssetId } from '@shapeshiftoss/caip'
 import { createSelector } from 'reselect'
 import type { ReduxState } from 'state/reducer'
+import {
+  selectEndpointNameParamFromFilter,
+  selectQueryStatusParamFromFilter,
+} from 'state/selectors'
 import { selectWalletId } from 'state/slices/common-selectors'
 
 import { nftApi } from './nftApi'
 import type { NftCollectionType, NftItem, NftItemWithCollection } from './types'
 
 const selectNfts = (state: ReduxState) => state.nft.nfts.byId
 const selectNftCollections = (state: ReduxState) => state.nft.collections.byId
+export const selectNftApiQueries = (state: ReduxState) => state.nftApi.queries
+
+export const selectNftApiQueriesByEndpointAndStatus = createSelector(
+  selectNftApiQueries,
+  selectEndpointNameParamFromFilter,
+  selectQueryStatusParamFromFilter,
+  (queries, endpointName, status) =>
+    Object.values(queries).filter(
+      query =>
+        (!endpointName || query?.endpointName === endpointName) &&
+        (!status || query?.status === status),
+    ),
+)
+
+export const selectGetNftUserTokensPending = createSelector(
+  (state: ReduxState) =>
```

Comment:
One could argue this is a dependency on the global `state` but this is the same we do in e.g swappers/selectors RTK queries selectors, and would have to be done either here or at component-level either way